lib/std/hash/benchmark.zig

```diff
@@ -203,13 +203,13 @@ pub fn benchmarkHashSmallKeysArrayPtr(
                 if (H.has_crypto_api) {
                     break :blk @truncate(u64, H.ty.toInt(small_key, init[0..H.ty.key_length]));
                 } else {
-                    break :blk H.ty.hash(init, small_key);
+                    break :blk @call(.never_inline, H.ty.hash, .{ init, small_key });
```

Comment:
Real-world applications are not going to use `.never_inline`, so the benchmark may not match actual applications.

Perhaps the actual issue is that in `small_key`, we inline too much and it doesn't fit in the L1/L2 cache any more.

Reply:
One can do cache simulation with valgrind (cachegrind in particular) and we might want to collect typical settings, possibly even in the wiki for baseline and more recent configurations:
https://valgrind.org/docs/manual/cg-manual.html "5.7.1. Cache Simulation Specifics"

Reply:
> Real-world applications are not going to use `.never_inline`, so the benchmark may not match actual applications.

I'm not quite sure this is right - I would have thought in real-world applications we don't know whether this call is going to get inlined or not. I did try changing the amount of explicit inlining in `xxhash.zig`, but it never really seemed to remove the need for this `@call(.never_inline...)` for best performance in the benchmark.

Reply:
The `.never_inline` calls have been removed.`